pghoard/pghoard.py

```diff
@@ -731,7 +734,9 @@ def run(self):
             except Exception as ex:  # pylint: disable=broad-except
                 self.log.exception("Unexpected exception in PGHoard main loop")
                 self.metrics.unexpected_exception(ex, where="pghoard_run")
-            time.sleep(5.0)
+            if self.thread_critical_failure_event.wait(timeout=5.0):
+                self.log.exception("Unexpected critical failure in PGHoard thread. Quitting main loop.")
```

Comment:
I'd send metrics data point here so that it's possible to escalate for this condition separately

Reply:
I added an additional metrics report inside the thread, where the event is set, because there's exception information available.